File: Exercism Codes/Java/Perfect Numbers/NaturalNumber.java
import java.util.stream.IntStream;

public class NaturalNumber {

    private final int number;

    public NaturalNumber(int number) {
        if(number < 1) throw new IllegalArgumentException("You must supply a natural number (positive integer)");
        this.number = number;
    }

    public Classification getClassification() {
        int aliquot = IntStream.range(1, number).filter(x -> number % x == 0).sum();
        return aliquot > number ? Classification.ABUNDANT : aliquot == number ? Classification.PERFECT : Classification.DEFICIENT;
    }
}

File: Exercism Codes/Kotlin/NaturalNumber.kt

enum class Classification {
    DEFICIENT, PERFECT, ABUNDANT
}

fun classify(naturalNumber: Int): Classification {
    if (naturalNumber > 0) {
        val sumOfFactors = (1 until naturalNumber).filter { naturalNumber % it == 0 }.sum()
        return if (sumOfFactors == naturalNumber) {
            Classification.PERFECT
        } else if (sumOfFactors > naturalNumber) {
            Classification.ABUNDANT
        } else Classification.DEFICIENT
    } else throw RuntimeException()
}
